src/main/java/org/spongepowered/api/item/inventory/BlockCarrier.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.item.inventory;
+
+import org.spongepowered.api.util.Direction;
+import org.spongepowered.api.world.Location;
+import org.spongepowered.api.world.World;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Represents a Block that is a carrier of an Inventory
+ * e.g. An Anvil
+ */
+public interface BlockCarrier extends Carrier {
+
+    /**
+     * Returns the Locations of the Blocks
+     *
+     * @return The Locations of the Blocks
+     */
+    List<Location<World>> getLocations();
```

Comment:
You've got this backwards. A BlockCarrier should be considered a single one by default. The sub interface MultiBlockCarrier should only be the one with multiple locations.

Reply:
Some of the mod multiblocks may have different inventories for different blocks in the multiblock structure. In this case, Zidane's idea prevails. You should also move the two methods below to `MultiBlockCarrier`.

Reply:
I'm not sure we can support mod multiblocks at all.
I don't see any concept of multiblock things in forge.